src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -105,14 +106,22 @@ public static void merge(List<RepoConfiguration> repoConfigs, List<AuthorConfigu
                 continue;
             }
 
-            matchingRepoConfig.setAuthorConfiguration(authorConfig);
+            if (!isDefaultBranch) {
+                matchingRepoConfig.setAuthorConfiguration(authorConfig);
+            }
         }
 
         for (AuthorConfiguration authorConfig : authorConfigs) {
             if (authorConfig.getLocation().isEmpty()) {
                 for (RepoConfiguration repoConfig : repoConfigs) {
                     repoConfig.addAuthors(authorConfig.getAuthorList());
                 }
+            } else if (authorConfig.isDefaultBranch()) {
```

Comment:
The logic here is acceptable. But we may have a helper functions like `getLocationMatchingRepoConfigs` to reduce the code complexity.